libs/pushstreams/gtest/src/PushStreamTestSuite.cc

```diff
@@ -0,0 +1,695 @@
+                        data = data + 1;
+                    }
+                    counter++;
+                }
+            });
+
+            t->join();
+            ses->close();
+            done.resolve();
+            return p;
+        };
+
+        auto x = ses->connectPromise().template then<void>(successLambda);
+
+        return ses;
+    }
+};
+
+TEST_F(PushStreamTestSuite, EventSourceCloseTest) {
+    int onClosedReceived{0};
+    int onErrorReceived{0};
+
+    auto ses = psp.template createSynchronousEventSource<int>();
+
+    auto successLambda = [](celix::Promise<void> p) -> celix::Promise<void> {
+        return p;
+    };
+    auto x = ses->connectPromise().then<void>(successLambda);
+    auto stream = psp.createUnbufferedStream<int>(ses);
+
+    auto streamEnded = stream->onClose([&](){
+        onClosedReceived++;
+    }).onError([&](){
+        onErrorReceived++;
+    }).forEach([&](int /*event*/) { });
+
+    ses->close();
+
+    streamEnded.wait();
+
+    ASSERT_EQ(1, onClosedReceived);
+    ASSERT_EQ(0, onErrorReceived);
+}
+
+TEST_F(PushStreamTestSuite, ChainedEventSourceCloseTest) {
+    int onClosedReceived{0};
+    int onErrorReceived{0};
+
+    auto ses = psp.template createSynchronousEventSource<int>();
+
+    auto successLambda = [](celix::Promise<void> p) -> celix::Promise<void> {
+        return p;
+    };
+    auto x = ses->connectPromise().then<void>(successLambda);
+
+    auto stream = psp.createUnbufferedStream<int>(ses);
+    auto& filteredStream = stream->filter([](const int& /*event*/) -> bool {
+            return true;
+        }).onClose([&](){
+            onClosedReceived++;
+        }).onError([&](){
+            onErrorReceived++;
+        });
+
+    auto streamEnded = filteredStream.forEach([&](int /*event*/) { });
+
+    ses->close();
+
+    streamEnded.wait();
+
+    ASSERT_EQ(1, onClosedReceived);
+    ASSERT_EQ(0, onErrorReceived);
+}
+
+
+TEST_F(PushStreamTestSuite, StreamCloseTest) {
+    int onClosedReceived{0};
+    int onErrorReceived{0};
+
+    auto ses = psp.createSynchronousEventSource<int>();
+
+    auto successLambda = [](celix::Promise<void> p) -> celix::Promise<void> {
+        return p;
+    };
+    auto x = ses->connectPromise().then<void>(successLambda);
+
+    auto stream = psp.createUnbufferedStream<int>(ses);
+    auto streamEnded = stream->onClose([&](){
+        onClosedReceived++;
+    }).onError([&](){
+        onErrorReceived++;
+    }).forEach([&](int /*event*/) { });
+
+    stream->close();
+
+    streamEnded.wait();
+
+    ASSERT_EQ(1, onClosedReceived);
+    ASSERT_EQ(0, onErrorReceived);
+}
+
+TEST_F(PushStreamTestSuite, PublishAfterStreamCloseTest) {
+    int onClosedReceived{0};
+    int onErrorReceived{0};
+    int onEventReceived{0};
+
+    auto ses = psp.createSynchronousEventSource<int>();
+
+    auto successLambda = [](celix::Promise<void> p) -> celix::Promise<void> {
+        return p;
+    };
+    auto x = ses->connectPromise().then<void>(successLambda);
```

Comment:
What does the `successLambda` and `x` do?

For this and the rest of the test cases

Reply:
They do nothing, dead code removed